CP-44638: determine network interface for private network

duobei opened the pull request:
Check networks that suit for private network in a pool:
1. network is managed
2. network is compatible with tunnel
3. pifs on network should satisfy below conditions:
    - pif is managed
    - tunnel is allowed on pif
    - there is no tunnel for XCP-NG using the pif
    - there is no other local pif on the network

gangj commented on ocaml/xapi/xapi_pif_helpers.ml:
```diff
@@ -177,15 +177,20 @@ let assert_tunnel_is_allowed_on_pif ~__context ~transport_PIF =
       false
-  | Tunnel_access _ :: _ ->
-      false
+  | Tunnel_access tunnel_ref :: _ -> (
```
I think the logic is not correct here: `Tunnel_access` is the pif created by `xapi_tunnel`, it should never be allowed to setup tunnel on it again.

gangj commented on ocaml/xapi/xapi_tunnel.ml:
```diff
@@ -94,3 +94,32 @@ let destroy ~__context ~self =
   Db.PIF.destroy ~__context ~self:pif ;
   Db.Tunnel.destroy ~__context ~self
+
+let determine_network_interface_for_private_network ~__context =
```
I think this might not be the expected work for "for determine Networkd interface". I think it is related to the design here: https://info.citrite.net/pages/viewpage.action?spaceKey=xenserver&title=Design%3A+XenServer+Network+Management+Features#Design:XenServerNetworkManagementFeatures-3.2.1InterfacesofNetworkd
For the change related to that design, I would suggest first to have a quick discussion to make sure we got the same understanding and have a local agreement first. As last for me, that part of design has changed, it is not as my previous understanding now.

psafont committed:
rrd_updates: output JSON in the same structure as XML

the data field is not part of meta. see the XML generator at
ocaml/libs/xapi-rrd/lib/rrd_updates.ml line 124

The issue was introduced when serialization was changed to use yojson:
https://github.com/xapi-project/xcp-rrd/commit/048b0e33b992e45ba8a465b24bbbfe10aa5d65c2

Signed-off-by: Pau Ruiz Safont <pau.ruizsafont@cloud.com>

psafont committed:
Merge pull request #5456 from psafont/metajson
